contracts/amm/guest/src/main.rs

```diff
@@ -22,23 +23,31 @@ fn main() {
 
     let amm_action = parsed_blob.data.parameters;
 
-    let mut callees_blob = Vec::new();
+    let mut callees_blobs = Vec::new();
     for blob in input.blobs.clone().into_iter() {
         if let Ok(structured_blob) = blob.data.clone().try_into() {
             let structured_blob: StructuredBlobData<Vec<u8>> = structured_blob; // for compiler
             if structured_blob.caller == Some(input.index.clone()) {
-                callees_blob.push(blob);
+                callees_blobs.push(blob);
             }
         };
     }
 
     let res = match amm_action {
-        AmmAction::Swap { from, pair } => amm_contract.verify_swap(callees_blob, from, pair),
+        AmmAction::Swap { from, pair } => amm_contract.verify_swap(callees_blobs, from, pair),
         AmmAction::NewPair {
             from,
             pair,
             amounts,
-        } => amm_contract.create_new_pair(from, pair, amounts, callees_blob),
+        } => {
+            let mut execution_state = ExecutionState { callees_blobs };
+            amm_contract
+                .create_new_pair(from, pair, amounts, &mut execution_state)
+                .and_then(|res| match execution_state.callees_blobs.is_empty() {
+                    true => Ok(res),
+                    false => Err(anyhow!("Callees blobs is not empty")),
```

Comment:
C'est pas plutôt 
```suggestion
                    false => Err(anyhow!("Callees blobs are empty")),
```
pour assurer qu'on a bien check les called blobs pour create_new pair ?

Reply:
Non justement il faut que ça soit empty, le process est qu'on setup les callee_blob et qu'on doit tous les appeler (on les enlève de la liste)

Mais c'est pas encore très clair